src/OpenTelemetry.Abstractions/Trace/AttributeValueBoolean.cs

```diff
@@ -0,0 +1,39 @@
+// <copyright file="AttributeValueBoolean.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Trace
+{
+    using System;
+
+    public abstract class AttributeValueBoolean : AttributeValue
```

Comment:
It may be way better to not expose those as public classes. `AttributeValue` class may be a factory for all types of attributes. I believe this will be a more natural way for developers to create attributes than thinking about the attribute type in advance.

Reply:
Makes sense, we can migrate these to `internal`.